checkpoint/orbax/checkpoint/_src/path/deleter.py

```diff
@@ -67,14 +68,54 @@ def __init__(
       directory: refer to CheckpointManager.directory
       todelete_subdir: refer to CheckpointManagerOptions.todelete_subdir
       name_format: refer to CheckpointManager._name_format
+      is_hns_enabled: refer to CheckpointManagerOptions.is_hns_enabled
       duration_metric: the name of the total delete duration metric
     """
     self._primary_host = primary_host
     self._directory = directory
     self._todelete_subdir = todelete_subdir
     self._name_format = name_format
+    self._is_hns_enabled = is_hns_enabled
     self._duration_metric = duration_metric
 
+  def _list_dirs_recursively(self, path: epath.Path):
+    """Lists all directories recursively under the given path."""
+    if path.is_dir():
+      yield path
+    for p in path.iterdir():
+      if p.is_dir():
+        yield from self._list_dirs_recursively(p)
+
+  def _rmtree(self, path: epath.Path):
```

Comment:
Can we also report a benchmark result on this?

Because based on my previous tests, I found that it's slow to delete them one-by-one.